Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1808,20 +1808,22 @@ WeakPtr<RenderBox> RenderGrid::getBaselineChild(ItemPosition alignment) const
     ASSERT(alignment == ItemPosition::Baseline || alignment == ItemPosition::LastBaseline);
     const RenderBox* baselineChild = nullptr;
     unsigned numColumns = currentGrid().numTracks(ForColumns);
-    unsigned numRows = currentGrid().numTracks(ForRows);
-
+    auto rowIndexDeterminingBaseline = alignment == ItemPosition::Baseline ? 0 : currentGrid().numTracks(ForRows) - 1;
     for (size_t column = 0; column < numColumns; column++) {
```

Comment:
It would be nice if this just iterated in the right direction, rather than calling `cell` twice and trying to swap the column number.

Does something like this work?

```
for (size_t column : alignment == ItemPosition::Baseline ? Range(0, numColumns) : makeReversedRange(Range(0, numCoumns))) {
    auto cell = currentGrid().cell(rowIndexDeterminingBaseline, column);

    for (auto& child : cell) {
        ....
}

```

Reply:
I don't think that quite works unfortunately since `Range` isn't a `Container` (but maybe there's something else that might make it work that I didn't notice). I went ahead and modified a logic a bit so we only call `cell` once though by choosing the column value based off of the alignment value